client/app.jsx

```diff
@@ -58,57 +60,91 @@ function App() {
     });
   }, []);
 
+  useEffect(() => {
+    const token = window.localStorage.getItem('react-context-jwt');
+    const user = token ? decodeToken(token) : null;
+    if (user) {
+      setUsername(user.nickname);
+      setUserId(user.userId);
+    }
+  }, []);
+
+  const handleSignIn = result => {
+    const { user, token } = result;
+    window.localStorage.setItem('react-context-jwt', token);
+    setUsername(user);
+  };
+
+  const handleSignOut = () => {
+    window.localStorage.removeItem('react-context-jwt');
+    setUsername(null);
+  };
+
+  const contextValue = {
+    username,
+    userId,
+    geoLocation,
+    setPageTitle,
+    pageTitle,
+    handleSignIn,
+    handleSignOut
+  };
   const renderPage = () => {
     return (
-      <Switch>
-        <Route exact path="/">
-          <TopAppBar />
-          <Home />
-          <BottomNavBar />
-        </Route>
-        <Route path="/new-post">
-          <TopAppBar />
-          <NewPost />
-          <BottomNavBar />
-        </Route>
-        <Route path="/chat">
-          <TopAppBar />
-          <Chat />
-          <BottomNavBar />
-        </Route>
-        <Route path="/favorites">
-          <TopAppBar />
-          <Favorites />
-          <BottomNavBar />
-        </Route>
-        <Route path="/profile">
-          <TopAppBar />
-          <Profile />
-          <BottomNavBar />
-        </Route>
-        <Route path="/search">
-          <TopAppBar />
-          <Search />
-        </Route>
-        <Route>
-          <Details path="/post" />
-        </Route>
-      </Switch>
+      <div>
+        {username
+          ? (
+          <div>
+            <Route exact path="/">
+              <TopAppBar />
+              <Home />
+              <BottomNavBar />
+            </Route>
+            <Route path="/new-post">
+              <TopAppBar />
+              <NewPost />
+              <BottomNavBar />
+            </Route>
+            <Route path="/chat">
+              <TopAppBar />
+              <Chat />
+              <BottomNavBar />
+            </Route>
+            <Route path="/favorites">
+              <TopAppBar />
+              <Favorites />
+              <BottomNavBar />
+            </Route>
+            <Route path="/profile">
+              <TopAppBar />
+              <Profile />
+              <BottomNavBar />
+            </Route>
+            <Route path="/search">
+              <TopAppBar />
+              <Search />
+            </Route>
+            <Route path="/post">
+              <Details />
+            </Route>
+          </div>
+            )
+          : (
+          <Route path="/auth">
+            <Auth />
+          </Route>
```

Comment:
Instead of a ternary, we'll normally use a redirect.

https://reactrouter.com/web/api/Redirect